plugins/azure/azure.plugin.zsh

```diff
@@ -0,0 +1,67 @@
+# AZ Get Subscritions
+function azgs() {
+  az account show --output tsv --query 'name'
+}
+
+# AZ Subscription Selection
+alias azss="az account set --subscription"
+
+
+function az_subscriptions() {
+  [[ -f "${AZURE_CONFIG_DIR:-$HOME/.azure}/accessTokens.json" ]] || return 1
+  az account list  --all --output tsv --query '[*].name'
+}
+
+function _az_subscriptions() {
+  reply=($(az_subscriptions))
+}
+compctl -K _az_subscriptions azss
+
+# Az prompt
+function prompt_azure() {
+  [[ ! -f "${AZURE_CONFIG_DIR:-$HOME/.azure/accessTokens.json}" ]] && return
+  # azgs is too expansive, if we have jq, we enable the prompt
+  [[ ! -f $(which jq) ]]  &&  return
+  azgs=$(cat ${AZURE_CONFIG_DIR:-$HOME/.azure/azureProfile.json} | jq -r '.subscriptions[] | select(.isDefault==true) .name')
+  echo "${ZSH_THEME_AZURE_PREFIX:=<az:}$(azgs)${ZSH_THEME_AZURE_SUFFIX:=>}"
+}
+
+if [ "$SHOW_AZ_PROMPT" != false ]; then
+  RPROMPT='$(prompt_azure)'"$RPROMPT"
```

Comment:
As a side, perhaps a question for the ohmyzsh maintainers, should new plugins generally be allowed to munge the prompt? It seems only three plugins actually do (`aws`, `fossil`, and `git-prompt` - though the last is specifically a plugin for the prompt). The rest rather document the `*_prompt_info()` and leave it up to the user and/or themes to customise. For instance [plugins/jenv/README.md](https://github.com/ohmyzsh/ohmyzsh/tree/0ab87c26c17171ae6162ff379a0c704fa57dff2e/plugins/jenv#theme-example), [plugins/virtualenv/README.md](https://github.com/ohmyzsh/ohmyzsh/tree/0ab87c26c17171ae6162ff379a0c704fa57dff2e/plugins/virtualenv#virtualenv)

Perhaps that munging behaviour is legacy in those plugins, and not removed for compatibility reasons?

Reply:
Absolutely, the plugin won't make assumption anymore on where it will show up, it will just load aliases, prepare functions and delegate the prompt to PROMPT or the theme explicitly.